deeplake/core/transform/transform_tensor.py

```diff
@@ -12,9 +14,40 @@ def __init__(self, name, dataset, items=None, slice_list=None) -> None:
         self.slice_list = slice_list or []
         self.length = None
         self._ndim = None
+        if not self.items:
+            self.items.append(True)
+
+    @property
+    def _numpy_only(self):
+        return self.items[0]
+
+    @_numpy_only.setter
+    def _numpy_only(self, val):
+        self.items[0] = val
+
+    def _indexed_numpy_array_list(self):
+        sub_sample_indexed = False
+        val = self.items[1:]
+        arr = False
+        for s in self.slice_list:
+            if isinstance(val, np.ndarray):
+                val = val[s]
+                if arr:
+                    sub_sample_indexed = True
+                arr = True
+            else:
+                val = slice_array_list(val, s)
+        if sub_sample_indexed:
+            self.length = 1
+            return [val]
+        else:
+            self.length = sum(map(lambda x: getattr(x, "__len__", lambda: 1)(), val))
+            return val
```

Comment:
This method is called only in numpy only mode. The return data is also in the batched form, but slicing is applied.